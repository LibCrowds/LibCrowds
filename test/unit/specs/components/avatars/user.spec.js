import BootstrapVue from 'bootstrap-vue'
import VueGravatar from 'vue-gravatar'

import pbTestResponses from '~/test/fixtures/pybossaTestResponses.json'
import testLocalConfig from '~/test/test.local.config'
import { mount, createLocalVue } from 'vue-test-utils'
import UserAvatar from '~/components/avatars/User'

describe('User avatar', () => {
  const localVue = createLocalVue()
  localVue.use(BootstrapVue)
  localVue.component('v-gravatar', VueGravatar)
  let user = pbTestResponses.getAccount.data.user
  const wrapper = mount(UserAvatar, {
    localVue,
    propsData: {
      user: user
    }
  })

  it('renders Gravatar when no custom avatar', () => {
    const renderer = require('vue-server-renderer').createRenderer()
    const src = wrapper.find('img').element.src
    expect(user.info.hasOwnProperty('avatar_url')).toBe(false)
    expect(src.startsWith('//www.gravatar.com')).toBe(true)
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('loads custom avatar', () => {
    const avatarUrl = 'http://example.com/avatar.jpg'
    wrapper.vm.loadAvatar(avatarUrl)
    expect(wrapper.vm.avatar).toBe(avatarUrl)
  })

  it('loads uploaded avatar', () => {
    const avatarUrl = '/uploads/user_1/img.jpg'
    const expectedUrl = testLocalConfig.pybossa.host + avatarUrl
    wrapper.vm.loadAvatar(avatarUrl)
    expect(wrapper.vm.avatar).toBe(expectedUrl)
  })
})
