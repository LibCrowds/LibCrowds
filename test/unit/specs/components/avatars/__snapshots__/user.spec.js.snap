// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User avatar renders Gravatar when no custom avatar 1`] = `

<div data-server-rendered="true"
     class="user-avatar"
>
  <img src="//www.gravatar.com/avatar/6579e96f76baa00787a28653876c6127?s=80&amp;d=identicon&amp;r=g"
       alt="Avatar for JohnDoe"
       triggers
       title="JohnDoe"
       class="img-thumbnail rounded-circle"
  >
</div>

`;
